PWGHF/HFL/Tasks/HFmAnalysis.cxx

```diff
@@ -0,0 +1,340 @@
+      // Before Muon Cuts
+      registry.fill(HIST("hMuBcuts"),
+                    values[VarManager::kPt],
+                    values[VarManager::kEta], dcaXY,
+                    values[VarManager::kCharge], track.p(),
+                    values[VarManager::kVtxZ],
+                    values[VarManager::kMuonTrackType],
+                    track.mcMask());
+
+      registry.fill(HIST("hPTBcuts"),
+                    values[VarManager::kPt], values[VarManager::kMCPt], values[VarManager::kMCPt] - values[VarManager::kPt], values[VarManager::kMuonTrackType]);
+      registry.fill(HIST("hEtaBcuts"),
+                    values[VarManager::kEta], values[VarManager::kMCEta], values[VarManager::kMCEta] - values[VarManager::kEta], values[VarManager::kMuonTrackType]);
+      registry.fill(HIST("hPBcuts"),
+                    values[VarManager::kP], values[VarManager::kMCPt] * std::cosh(values[VarManager::kMCEta]), values[VarManager::kMCPt] * std::cosh(values[VarManager::kMCEta]) - values[VarManager::kP], values[VarManager::kMuonTrackType]);
+      // After Muon Cuts
+      if (trackCut->IsSelected(values)) {
+        registry.fill(HIST("hMuAcuts"),
+                      values[VarManager::kPt],
+                      values[VarManager::kEta], dcaXY,
+                      values[VarManager::kCharge], track.p(),
+                      values[VarManager::kVtxZ],
+                      values[VarManager::kMuonTrackType],
+                      track.mcMask());
+
+        registry.fill(HIST("hPTAcuts"),
+                      values[VarManager::kPt], values[VarManager::kMCPt], values[VarManager::kMCPt] - values[VarManager::kPt], values[VarManager::kMuonTrackType]);
+        registry.fill(HIST("hEtaAcuts"),
+                      values[VarManager::kEta], values[VarManager::kMCEta], values[VarManager::kMCEta] - values[VarManager::kEta], values[VarManager::kMuonTrackType]);
+        registry.fill(HIST("hPAcuts"),
+                      values[VarManager::kP], values[VarManager::kMCPt] * std::cosh(values[VarManager::kMCEta]), values[VarManager::kMCPt] * std::cosh(values[VarManager::kMCEta]) - values[VarManager::kP], values[VarManager::kMuonTrackType]);
+      }
+    } // end loop over muon tracks
+  }
+
+  void processMuonDataAO2D(MyEventsSelected::iterator const& event, aod::BCs const& bcs,
+                           aod::FullFwdTracks const& tracks)
+  {
+    runDataMuonSel<gEventFillMap, gMuonFillMap>(event, bcs, tracks);
+  }
+
+  void processMuonMCAO2D(MyMcEventsSelected::iterator const& event, aod::BCs const& bcs,
+                         MyMuons const& tracks, aod::McParticles const& mc)
+  {
+    runMCMuonSel<gEventFillMap, gMuonFillMap, gTrackMCFillMap>(event, bcs, tracks, mc);
+  }
+
+  PROCESS_SWITCH(MuonSelection, processMuonDataAO2D, "run muonselection with Data AO2D file", true);
+  PROCESS_SWITCH(MuonSelection, processMuonMCAO2D, "run muonselection with MC AO2D file", false);
```

Comment:
Thank you for the comments. The original purpose of the "AO2D" suffix is to distinguish between AO2D input and reducedAOD input which may be added in the future. I will fix them since it can not be applied to reduced AOD files at present, but I probably need to change them again afterwards.

Reply:
OK, I see. Thanks for the clarification. In that case, you can for example add a suffix `Reduced` to the names of the new functions when you add them.